restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmConfigurationHandler.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2017, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ * For questions related to commercial use licensing, please contact sales@telestax.com.
+ *
+ */
+
+package org.restcomm.android.sdk.fcm;
+
+import com.google.firebase.iid.FirebaseInstanceId;
+
+import org.restcomm.android.sdk.RCClient;
+import org.restcomm.android.sdk.RCDevice;
+import org.restcomm.android.sdk.fcm.model.FcmApplication;
+import org.restcomm.android.sdk.fcm.model.FcmBinding;
+import org.restcomm.android.sdk.fcm.model.FcmCredentials;
+import org.restcomm.android.sdk.storage.StorageManager;
+import org.restcomm.android.sdk.util.RCLogger;
+
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Pair;
+
+import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.HashSet;
+
+/**
+ * Manage logic for registering device and user for push notifications
+ *
+ */
+
+public class FcmConfigurationHandler {
+
+    private static final String TAG = FcmConfigurationHandler.class.getCanonicalName();
+
+    private static final String FCM_ACCOUNT_SID = "fcm-account-sid";
+    private static final String FCM_CLIENT_SID = "fcm-client-sid";
+    private static final String FCM_APPLICATION = "fcm-application";
+    private static final String FCM_CREDENTIALS = "fcm-credentials";
+    private static final String FCM_BINDING = "fcm-binding";
+
+    private static final String TYPE = "fcm";
+
+    private WeakReference<StorageManager> mStorageManagerWeak;
```

Comment:
Do we really need WeakReferences here? I mean the lifetime of FcmConfigurationHandler is the lifetime of the thread running underneath the Async Task, so we 're guaranteed that it will be freed after the Async Task finishes. Am I missing something?

Reply:
I put it because of Context reference that StorageManagerPreferences is using for saving all data.